Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -33,6 +33,13 @@ def change_rpath(file, old_prefix, new_prefix)
       lib_path = "#{new_prefix}/lib"
       rpath << lib_path unless rpath.include? lib_path
 
+      # Add GCC's lib directory (as of GCC 12+) to RPATH when there is existing linkage.
+      # This fixes linkage for newly-poured bottles.
+      # TODO: Replace with
+      #   rpath.map! { |path| path = path.sub(%r{lib/gcc/\d+$}, "lib/gcc/current") }
+      # when Homebrew/homebrew-core#106755 is merged.
+      rpath.prepend HOMEBREW_PREFIX/"opt/gcc/lib/gcc/current" if rpath.any? { |rp| rp.match?(%r{lib/gcc/\d+$}) }
```

Comment:
Should `%r{lib/gcc/\d+$}` check to see if this is in the Cellar/opt/etc.?

Reply:
This is a good question. I wanted to say that this should always be `opt`, so there is no need to check, but there is one possible exception: the GCC formulae themselves.

I'm now wondering if we should exclude the GCC formulae from this. I'm leaning toward yes.